### PR TITLE
Only stash blobs when closing

### PR DESCRIPTION
**Pull request by dannimad**
This change makes getPendingLocalState to stash blobs only in closeAndGetPendingLocalState case (notifyImminentClosure = true). getPendingLocalState without closing can't stash blobs given that their reference could not be included in the pending state, having as a result a blob handle without any reference, dds that uses it. 

Some tests relied on being able to read pending attachment blobs which is not possible now, so I changed some asserts and ways to confirm correct behavior. There is one test we're losing here "blob is acked after upload". Given that we can't collect pending blobs if they are not attached, there is no way to know if the stashed blob is purely acked.  I substituted that test with "removes pending blob when waiting for blob to be attached" which indirectly checks the same thing, given that only blobs who are attacked and acked are removed.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 1]
```diff
@@ -1268,6 +1273,30 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 		);
 	});
 
+	it("get pending blob without close while online", async function () {
```
Comment (wes-carlson) [inline packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:1276]: What's being tested here? What you're doing is something we don't want people to do (using stash blob when the original container hasn't been closed yet)

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 2]
```diff
@@ -1022,6 +1022,9 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 
 			[...Array(lots).keys()].map((i) => dataStore.root.set(`test op #${i}`, i));
 
+			// wait for flush to happen and see added ops in the pending queue
```
[the remark this answers is not in the document]
Comment (dannimad) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:1025]: If they are not flushed, they are not yet added to the pending list in the pending state manager. I could move the flush inside getPendingLocalState to happen before getting local state

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts (hunk 1 above)
Comment (dannimad) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:1276]: Got it, I'll remove this test then.

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (wes-carlson) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts]: I think that would make sense. I'm not sure there's any point to flushing afterwards.

**Review by wes-carlson — APPROVED** [review]

**Pull request merged**